crates/sqlexec/src/remote/client.rs

```diff
@@ -173,15 +157,18 @@ impl RemoteClient {
     }
 
     /// Connect to a proxy destination.
-    pub async fn connect_with_proxy_destination(dst: ProxyDestination) -> Result<Self> {
-        Self::connect_with_proxy_auth_params(dst.dst.to_string(), dst.params, dst.tls_conf).await
+    pub async fn connect_with_proxy_destination(
```

Comment:
my inclination would be to do the query for the cert in the main() if we're running glaredb in a mode where it _might_ do hybrid exec. (i.e. when we construct the remote client) so that, in the case that this errors, we would get that error very early rather than late.

Reply:
The reason for doing so here is relative to the common call from Python and Local. I can take a look at lifting it up, but I think it means changing the Python logic and I wasn't wanting to do that

Reply:
makes sense. I think it probably still makes sense, but definitely can save that kind of refactoring for later.